tests/utils/test_price_utils.py

```diff
@@ -0,0 +1,577 @@
+                "token_asset_reserves": 200000, # PT reserves; y = 200000
+                "days_remaining": 91.25, # 3 months remaining; t = 0.25
+                "time_stretch": 36.97812836141986, # 36.97812836141986 time_stretch; T = 0.006760753209
+                "init_share_price": 1.5, # original share price pool started; u = 1.5
+                "share_price": 2, # share price of the LP in the yield source; c = 2
+                # from the inputs, we have:
+                # c = 2
+                # y = 500000
+                # u = 1.5
+                # r = 0.01
+                # t = 0.25
+                # T = 0.006760753209
+                # x = 2*c*y/(u*(r*t + 1)**(1/T) - c)
+                #   = 2*2*200000/(1.5*(0.01*0.25 + 1)**(1/0.006760753209) - 1)
+                #   = 4702414.821874424
+                "expected_result": 4702414.821874424 # base > token reserves (too much? share_price sus)
+            },
+            # test 6: 6% APR; 800k bond reserves; 3mo remaining;
+            #   36.97812836141986 t_stretch (targets 3% APR);
+            #   1.5 init share price; 2 share price
+            {
+                "apr": 0.06, # fixed rate APR you'd get from purchasing bonds; r = 0.06
+                "token_asset_reserves": 800000, # PT reserves; y = 800000
+                "days_remaining": 91.25, # 3 months remaining; t = 0.25
+                "time_stretch": 36.97812836141986, # 36.97812836141986 time_stretch; T = 0.006760753209
+                "init_share_price": 1.5, # original share price pool started; u = 1.5
+                "share_price": 2, # share price of the LP in the yield source; c = 2
+                # from the inputs, we have:
+                # c = 2
+                # y = 800000
+                # u = 1.5
+                # r = 0.06
+                # t = 0.25
+                # T = 0.006760753209
+                # x = 2*c*y/(u*(r*t + 1)**(1/T) - c)
+                #   = 2*2*800000/(1.5*(0.06*0.25 + 1)**(1/0.006760753209) - 1)
+                #   = 276637.1374102353
+                "expected_result": 276637.1374102353 # token > base reserves (is it enough? share_price sus)
+            }
+        ]
+
+        for test_case in test_cases:
+            base_asset_reserves = price_utils.calc_base_asset_reserves(
+                test_case["apr"],
+                test_case["token_asset_reserves"],
+                test_case["days_remaining"],
+                test_case["time_stretch"],
+                test_case["init_share_price"],
+                test_case["share_price"]
+            )
+
+            np.testing.assert_almost_equal(
+                base_asset_reserves,
+                test_case["expected_result"],
+                err_msg="unexpected base_asset_reserves",
+            )
+
+
+    def test_calc_liquidity(self):
+        """Unit tests for the calc_liquidity function"""
+
+        test_cases = [
```

Comment:
same here, let's try to break the function with crazy APR, market_price, share_price

Reply:
I think this is resolved? @sentilesdal or @MazyGio can you resolve it if you're happy?